.github/workflows/build-test-ship.yaml

```diff
@@ -131,21 +134,25 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
 
-      # On push/PR, ensure service images can be built on target platforms.
-      - name: Build final images
+      # Build and upload image artifact for use in api-proctor tests.
+      - name: Build image for api-proctor tests
         uses: docker/build-push-action@v2
-        if: |
-          github.event_name == 'pull_request' ||
-          (github.event_name == 'push' &&
-           steps.check-version-change.outputs.appchange == 'false')
         with:
           context: .
           file: ./cmd/${{ matrix.service }}/Dockerfile
           cache-from: type=gha
           cache-to: type=gha,mode=max
-          platforms: linux/amd64,linux/arm64
+          outputs: type=docker,dest=${{ matrix.service }}.tar
+          tags: ${{ matrix.service }}:ci
 
-      # On release, build service images and push to ECR.
+      - name: Upload docker image artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.service }}
+          path: ${{ matrix.service }}.tar
+          retention-days: 1
```

Comment:
Since jobs run on different machines (unless you're using custom runners), we can't trivially access files produced in one job from another. We have to upload the files in one job and download them in another, which we can do using GitHub artifacts. I've set the retention period to the lowest possible (1 day). See the PR writeup for caveats.

Ideally, we'd just build the services images in the `run-api-proctor-tests` job itself, to avoid the need for this inter-job communication. However, I'm not sure that's feasible because of the way we're using the `matrix` strategy to build all services. If we used a `matrix` strategy in `run-api-proctor-tests`, then instead of running the tests once with all _N_ of the branch's services running, we'd end up running the tests _N_ times - and each of those runs would only be using the branch's version of _one_ service at a time.